database/migration/src/main/resources/changesets/change_table_owner_for_workflows_app_cromwell.xml

```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog objectQuotingStrategy="QUOTE_ALL_OBJECTS"
+                   xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
+
+    <!-- The 'sharedCromwellDbRole' will be set via JAVA_OPTS in Cromwell helm chart in terra-helmfile. -->
+    <property  name="sharedCromwellDbRole" value=""/>
+
+    <!--
+        This changeset will be only applied when 'sharedCromwellDbRole' property is set and the role exists
+        in the database. Currently, it is only set for the shared Cromwell service in the WORKFLOWS app.
+        For all the other Cromwell servers this should be a no-op.
+     -->
+    <changeSet author="sshah" id="change_table_owner_for_workflows_app_cromwell" dbms="postgresql">
+        <preConditions onFail="MARK_RAN">
+            <!-- check that 'sharedCromwellDbRole' role is not null and exists in the roles table -->
+            <sqlCheck expectedResult="1">
+                SELECT count(*)
+                FROM pg_roles
+                where '${sharedCromwellDbRole}' is NOT NULL AND rolname = '${sharedCromwellDbRole}';
+            </sqlCheck>
+            <!-- check that table ownership update hasn't already happened -->
+            <sqlCheck expectedResult="0">
+                SELECT count(*)
+                FROM pg_tables
+                WHERE tableowner = '${sharedCromwellDbRole}';
+            </sqlCheck>
```

Comment:
This seems superfluous and potentially unwanted. Say somehow we end up with some other unrelated table owned by that role when we want to run the update... wouldn't it be better to run it anyway, rather than MARK_RAN'ing it and not doing the update?

Reply:
I will remove this check 👍